kufar/books/main.go

```diff
@@ -208,10 +209,22 @@ func (d Database) SignInUser(w http.ResponseWriter, r *http.Request) {
 
 	cred.Username = credReq.Username
 
-	session.Values["user"] = cred.UUID
-	if err = session.Save(r, w); err != nil {
-		http.Error(w, http.StatusText(http.StatusInternalServerError)+": could not assign cookie", http.StatusInternalServerError)
-		return
+	credentials := r.Header.Get("Authorization")
+
+	switch {
+	case credentials != "":
+		r.SetBasicAuth(cred.Username, cred.Password)
+		auth := strings.SplitN(r.Header.Get("Authorization"), " ", 2)
```

Comment:
Here, you are doing nothing with your data, just validate it and nothing more.

Reply:
not sure I got you. im validating what I got from the Authorization header here(checking if its the Basic type, and the length also). I can validate the users credentials, but that'll require me to decode the base64 first